### PR TITLE
V2.1 spring security upgrade

### PR DESCRIPTION
**Pull request by hilpitome**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,6 +28,7 @@
 		<opensrp.connector.version>2.4.1-SNAPSHOT</opensrp.connector.version>
         <opensrp.interface.version>2.0.1-SNAPSHOT</opensrp.interface.version>
         <powermock.version>2.0.5</powermock.version>
+        <opensrp-server-openlmis-stock-version>1.1.2-SNAPSHOT</opensrp-server-openlmis-stock-version>
```
Comment (ekigamba) [inline pom.xml:31]: This is supposed to be an update of dependencies. Why was this added?

File: pom.xml [hunk 2]
```diff
@@ -429,7 +462,13 @@
...
                     <target>11</target>
                     <encoding>UTF-8</encoding>
-
+                    <annotationProcessorPaths>
+                        <path>
+                            <groupId>org.projectlombok</groupId>
+                            <artifactId>lombok</artifactId>
+                            <version>${lombok.version}</version>
+                        </path>
+                    </annotationProcessorPaths>
```
Comment (ekigamba) [inline pom.xml:471]: Should all these dependencies be added? Is this a mistake of cherry-picking from the master server-web? Kindly check

File: pom.xml [hunk 3]
```diff
@@ -799,5 +842,30 @@
...
+            <properties>
+                <openlmis.stock.profile>openlmis-stock</openlmis.stock.profile>
+            </properties>
+        </profile>
+        <profile>
+            <id>plan-generation</id>
+            <properties>
+                <plan.generation.profile>plan-generation</plan.generation.profile>
+            </properties>
+        </profile>
```
Comment (ekigamba) [inline pom.xml:869]: These also

File: src/main/java/org/opensrp/web/GzipResponseStream.java [hunk 4]
```diff
@@ -75,4 +76,17 @@ public boolean closed() {
...
+	public boolean isReady() {
+		return true;
+	}
+
+	@Override
+	public void setWriteListener(WriteListener writeListener) {
+		try {
+			writeListener.onWritePossible();
+		} catch (IOException e){
+			writeListener.onError(e);
```
Comment (ekigamba) [inline src/main/java/org/opensrp/web/GzipResponseStream.java:89]: Let's log this also

File: src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java [hunk 5]
```diff
@@ -108,15 +109,38 @@ private static boolean isValidJSON(final String json) throws IOException {
...
+		public boolean isReady() {
+			return true;
+		}
+
+		@Override
+		public void setReadListener(ReadListener readListener) {
+			try {
+				readListener.onDataAvailable();
+			} catch (IOException e) {
+				readListener.onError(e);
```
Comment (ekigamba) [inline src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java:141]: Log this error

**Review by ekigamba — CHANGES_REQUESTED** [review]

File: src/main/java/org/opensrp/web/GzipResponseStream.java (hunk 4 above)
Comment (hilpitome) [reply src/main/java/org/opensrp/web/GzipResponseStream.java:89]: noted

File: src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java (hunk 5 above)
Comment (hilpitome) [reply src/main/java/org/opensrp/web/config/security/filter/XssPreventionRequestWrapper.java:141]: noted as well

File: pom.xml (hunk 2 above)
Comment (hilpitome) [reply pom.xml:471]: will check

Comment (ekigamba) [reply pom.xml:471]: Kindly check this again

**Review by ekigamba — CHANGES_REQUESTED** [review]

**Review by ekigamba — APPROVED** [review]
